Client/AdjustTelemetryHelper.swift

```diff
@@ -0,0 +1,62 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Foundation
+import Adjust
+import Glean
+
+protocol AdjustTelemetryData {
+    var campaign: String? { get set }
+    var adgroup: String? { get set }
+    var creative: String? { get set }
+    var network: String? { get set }
+}
+
+extension ADJAttribution: AdjustTelemetryData {
+}
+
+protocol AdjustTelemetryProtocol {
+    func setAttribution(_ attribution: AdjustTelemetryData?)
+    func sendDeeplinkTelemetry(url: URL)
+}
+
+class AdjustTelemetryHelper: AdjustTelemetryProtocol {
+
+    // MARK: - UserDefaults
+
+    private enum UserDefaultsKey: String {
+        case hasSetAttributionData = "com.moz.adjust.hasSetAttributionData.key"
+    }
+
+    var hasSetAttributionData: Bool {
+        get { UserDefaults.standard.object(forKey: UserDefaultsKey.hasSetAttributionData.rawValue) as? Bool ?? false }
+        set { UserDefaults.standard.set(newValue, forKey: UserDefaultsKey.hasSetAttributionData.rawValue) }
+    }
+
+    func sendDeeplinkTelemetry(url: URL) {
+        let extra = [TelemetryWrapper.EventExtraKey.deeplinkURL.rawValue: url.absoluteString]
+
+        TelemetryWrapper.recordEvent(category: .action,
+                                     method: .applicationOpenUrl,
+                                     object: .deeplinkReceived,
+                                     value: nil,
+                                     extras: extra)
+    }
+
+    func setAttribution(_ attribution: AdjustTelemetryData?) {
+        guard !hasSetAttributionData else { return }
```

Comment:
Once set, this telemetry is saved in Glean and we don't have to set it again? i.e let's say we restart the app, then it's still there in Glean?

Reply:
that's a good question because it's called on the `adjustAttributionChanged(_ attribution: ADJAttribution?)` delegate I added hasSetAttributionData to handle the case of existing users. I will ask what's the expected behaviour